### PR TITLE
[DO NOT MERGE]Replacing usage of MRN

### PR DESCRIPTION
**Pull request by alexflav23**

### CODE REVIEW

**Review by amaalali — CHANGES_REQUESTED** [review]
A few more thing that we need to address
- I don't think this works at the current moment, the data would be saved correctly but the user journey would immediately break on the next page. The SessionRepository would need to be updated since we now need to fetch by the `ArrivalUniqueRef` instead of the MRN (which is saved in the _id field)
- To protect the Session repository from these sort of breakages in the future I suggest that we change the `SessionRepository`'s `get` to be `def get(id: ArrivalUniqueRef, eoriNumber: EoriNumber): Future[Option[UserAnswers]]` 
- We'll need to add an index for the `ref` field in user-answers collection since this is used by the session repository for it's query. I would also recommend making this a unique.

File: app/models/ArrivalUniqueRef.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+package models
+
+import java.util.UUID
+
+import play.api.libs.json.{Json, OFormat}
+import play.api.mvc.PathBindable
+
+import scala.util.{Failure, Success, Try}
+final case class ArrivalUniqueRef(uuid: UUID) {
```
Comment (amaalali) [inline app/models/ArrivalUniqueRef.scala:25]: Would `DraftArrivalRef` be a better name for this? It's not an arrival message as yet, just a in-progress draft at this point

File: app/models/UserAnswers.scala [hunk 2]
```diff
@@ -91,7 +95,8 @@ object UserAnswers {
         (__ \ "eoriNumber").write[EoriNumber] and
         (__ \ "data").write[JsObject] and
         (__ \ "lastUpdated").write(MongoDateTimeFormats.localDateTimeWrite) and
-        (__ \ "arrivalId").writeNullable[ArrivalId]
+        (__ \ "arrivalId").writeNullable[ArrivalId] and
+        (__ \ "ref").write[ArrivalUniqueRef]
```
Comment (amaalali) [inline app/models/UserAnswers.scala:99]: This Json Writes would result in a Json shape of:

```
{
    "_id" : "99IT9876AB88901209",
    "data" : {},
    "eoriNumber" : "GB1212",
    "lastUpdated" : ISODate("2020-09-07T12:01:42.109Z"),
    "ref" : {
        "uuid" : "893dca38-a174-4aa6-abc4-cc1ab158cd67"
    }
}
```

But I think this should be flattened out like:

```
{
    "_id" : "99IT9876AB88901209",
    "data" : {},
    "eoriNumber" : "GB1212",
    "lastUpdated" : ISODate("2020-09-07T12:01:42.109Z"),
    "draftArrivalRef" : "893dca38-a174-4aa6-abc4-cc1ab158cd67"
}
```

**Review by amaalali — CHANGES_REQUESTED** [review]

File: .g8/checkboxPage/migrations/$className__snake$.sh [hunk 3]
```diff
@@ -6,10 +6,10 @@ echo "Applying migration $className;format="snake"$"
 echo "Adding routes to conf/app.routes"
 
 echo "" >> ../conf/app.routes
-echo "GET        /:mrn/$className;format="decap"$                        controllers.$className$Controller.onPageLoad(mrn: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
+echo "GET        /:mrn/$className;format="decap"$                        controllers.$className$Controller.onPageLoad(ref: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
```
Comment (chris-ianson) [inline .g8/checkboxPage/migrations/$className__snake$.sh:9]: First argument needs to change to ArrivalUniqueRef

File: .g8/checkboxPage/migrations/$className__snake$.sh [hunk 4]
```diff
@@ -6,10 +6,10 @@ echo "Applying migration $className;format="snake"$"
...
 echo "POST       /:mrn/$className;format="decap"$                        controllers.$className$Controller.onSubmit(mrn: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
 
-echo "GET        /:mrn/change$className$                  controllers.$className$Controller.onPageLoad(mrn: MovementReferenceNumber, mode: Mode = CheckMode)" >> ../conf/app.routes
+echo "GET        /:mrn/change$className$                  controllers.$className$Controller.onPageLoad(ref: MovementReferenceNumber, mode: Mode = CheckMode)" >> ../conf/app.routes
```
Comment (chris-ianson) [inline .g8/checkboxPage/migrations/$className__snake$.sh:12]: First argument needs to change to ArrivalUniqueRef

File: .g8/checkboxPage/migrations/$className__snake$.sh [hunk 5]
```diff
@@ -6,10 +6,10 @@ echo "Applying migration $className;format="snake"$"
...
 echo "POST       /:mrn/change$className$                  controllers.$className$Controller.onSubmit(mrn: MovementReferenceNumber, mode: Mode = CheckMode)" >> ../conf/app.routes
```
Comment (chris-ianson) [inline .g8/checkboxPage/migrations/$className__snake$.sh:13]: First argument needs to change to ArrivalUniqueRef

File: .g8/datePage/migrations/$className__snake$.sh [hunk 6]
```diff
@@ -6,10 +6,10 @@ echo "Applying migration $className;format="snake"$"
 echo "Adding routes to conf/app.routes"
 
 echo "" >> ../conf/app.routes
-echo "GET        /:mrn/$className;format="decap"$                  controllers.$className$Controller.onPageLoad(mrn: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
+echo "GET        /:mrn/$className;format="decap"$                  controllers.$className$Controller.onPageLoad(ref: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
 echo "POST       /:mrn/$className;format="decap"$                  controllers.$className$Controller.onSubmit(mrn: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
 
-echo "GET        /:mrn/change$className$                        controllers.$className$Controller.onPageLoad(mrn: MovementReferenceNumber, mode: Mode = CheckMode)" >> ../conf/app.routes
+echo "GET        /:mrn/change$className$                        controllers.$className$Controller.onPageLoad(ref: MovementReferenceNumber, mode: Mode = CheckMode)" >> ../conf/app.routes
```
Comment (chris-ianson) [inline .g8/datePage/migrations/$className__snake$.sh:12]: First argument needs to change to ArrivalUniqueRef

File: .g8/datePage/migrations/$className__snake$.sh [hunk 7]
```diff
@@ -6,10 +6,10 @@ echo "Applying migration $className;format="snake"$"
 echo "Adding routes to conf/app.routes"
 
 echo "" >> ../conf/app.routes
-echo "GET        /:mrn/$className;format="decap"$                  controllers.$className$Controller.onPageLoad(mrn: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
+echo "GET        /:mrn/$className;format="decap"$                  controllers.$className$Controller.onPageLoad(ref: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
```
Comment (chris-ianson) [inline .g8/datePage/migrations/$className__snake$.sh:9]: First argument needs to change to ArrivalUniqueRef

File: .g8/datePage/migrations/$className__snake$.sh [hunk 8]
```diff
@@ -6,10 +6,10 @@ echo "Applying migration $className;format="snake"$"
...
 echo "POST       /:mrn/$className;format="decap"$                  controllers.$className$Controller.onSubmit(mrn: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
```
Comment (chris-ianson) [inline .g8/datePage/migrations/$className__snake$.sh:10]: First argument needs to change to ArrivalUniqueRef

File: .g8/datePage/migrations/$className__snake$.sh [hunk 9]
```diff
@@ -6,10 +6,10 @@ echo "Applying migration $className;format="snake"$"
...
 
-echo "GET        /:mrn/change$className$                        controllers.$className$Controller.onPageLoad(mrn: MovementReferenceNumber, mode: Mode = CheckMode)" >> ../conf/app.routes
+echo "GET        /:mrn/change$className$                        controllers.$className$Controller.onPageLoad(ref: MovementReferenceNumber, mode: Mode = CheckMode)" >> ../conf/app.routes
 echo "POST       /:mrn/change$className$                        controllers.$className$Controller.onSubmit(mrn: MovementReferenceNumber, mode: Mode = CheckMode)" >> ../conf/app.routes
```
Comment (chris-ianson) [inline .g8/datePage/migrations/$className__snake$.sh:13]: First argument needs to change to ArrivalUniqueRef

File: .g8/intPage/migrations/$className__snake$.sh [hunk 10]
```diff
@@ -6,10 +6,10 @@ echo "Applying migration $className;format="snake"$"
 echo "Adding routes to conf/app.routes"
 
 echo "" >> ../conf/app.routes
-echo "GET        /:mrn/$className;format="decap"$                  controllers.$className$Controller.onPageLoad(mrn: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
+echo "GET        /:mrn/$className;format="decap"$                  controllers.$className$Controller.onPageLoad(ref: MovementReferenceNumber, mode: Mode = NormalMode)" >> ../conf/app.routes
```
Comment (chris-ianson) [inline .g8/intPage/migrations/$className__snake$.sh:9]: First argument needs to change to ArrivalUniqueRef

**Review by chris-ianson — CHANGES_REQUESTED** [review]
The g8 templates that we use to build different pages are still referencing MRN so when new pages are created they won't work. Could we update all the templates to use ArrivalUniqueRef rather than MovemmentReferenceNumber

Also the following tests are failing:
- ArrivalRejectionControllerSpec

**Pull request closed without merging**